### PR TITLE
prom-wrapper: introduce src-prometheus, migrate alerts history

### PR DESCRIPTION
**Pull request by bobheadxi**
Was doing a review of the prom-wrapper code and remembered this old code I wrote lying around for the frontend making a direct query to Prometheus, which powers the alerts overview in the bug reports page. There's a few very manual API requests to the prom-wrapper as well. Sadly it seems I wrote some pretty bad code here and figured it might be a good idea to get this cleaned up a bit. Then I got into it and updated _all_ the frontend->prom-wrapper touch points (more than I recalled) into a new package, `src-prometheus`

This PR:

- removes the frontend->prometheus direct querying, and lets the prom-wrapper handle it (it already provides aggregated overviews for `alerts.hideObservabilitySiteAlerts`)
- introduces the `src-prometheus` package for the prom-wrapper API, and replaces all touch points with the prom-wrapper with usage of this API
- leverage the `src-prometheus` package in the prom-wrapper

Post-this PR I would advocate to move `prom-wrapper` out of `docker-images` and into a top-level `cmd/prom-wrapper` since doing this I realized there's a lot more to prom-wrapper than just a reverse-proxy for alertmanager and prometheus

### CODE REVIEW

**Comment by uwedeportivo** [comment]
i'll look at this tomorrow morning

File: cmd/frontend/graphqlbackend/site_alerts.go [hunk 1]
```diff
@@ -107,8 +104,11 @@ func init() {
 	// Notify when updates are available, if the instance can access the public internet.
 	AlertFuncs = append(AlertFuncs, updateAvailableAlert)
 
-	// Notify admins if critical alerts are firing.
-	AlertFuncs = append(AlertFuncs, observabilityActiveAlertsAlert(prometheusutil.PrometheusURL))
+	// Notify admins if critical alerts are firing, if Prometheus is configured.
+	promURL, err := srcprometheus.PrometheusURL()
```
Comment (uwedeportivo) [inline cmd/frontend/graphqlbackend/site_alerts.go:108]: i found this slightly weird to read. could `srcprometheus.PrometheusURL()` instead return the empty string when prometheus is not configured ?

**Review by uwedeportivo — APPROVED** [review]
LGTM

avoiding the dash for the package name is unfortunate :-), but i cannot come up with a different name

File: cmd/frontend/graphqlbackend/site_alerts.go (hunk 1 above)
Comment (bobheadxi) [reply cmd/frontend/graphqlbackend/site_alerts.go]: tried to improve the flow here a bit with https://github.com/sourcegraph/sourcegraph/pull/17602/commits/841ac9c70b36c654f4c89cfa6d724a4520d35f79, now usages should just directly create a client and the client creation will return the appropriate ErrPrometheusNotAvailable that can be checked!

**Review by uwedeportivo — APPROVED** [review]

File: cmd/frontend/graphqlbackend/site_alerts.go (hunk 1 above)
Comment (uwedeportivo) [reply cmd/frontend/graphqlbackend/site_alerts.go]: nice, lgtm

**Comment by bobheadxi** [comment]
Did a final sanity check + some tweaks:

image

**Pull request merged**